modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/AbstractPythonCodegen.java

```diff
@@ -114,6 +119,35 @@ public void processOpts() {
             LOGGER.info("Environment variable PYTHON_POST_PROCESS_FILE not defined so the Python code may not be properly formatted. To define it, try 'export PYTHON_POST_PROCESS_FILE=\"/usr/local/bin/yapf -i\"' (Linux/Mac)");
             LOGGER.info("NOTE: To enable file post-processing, 'enablePostProcessFile' must be set to `true` (--enable-post-process-file for CLI).");
         }
+
+        if (Boolean.TRUE.equals(additionalProperties.get(CodegenConstants.SUPPORT_PYTHON39))) {
+            additionalProperties.put(CodegenConstants.SUPPORT_PYTHON39, Boolean.TRUE);
+
+            // Ref: https://mypy.readthedocs.io/en/stable/cheat_sheet_py3.html
+            // update type mapping for type hints
+            typeMapping.put("set", "set");
+            typeMapping.put("array", "list");
+            typeMapping.put("map", "dict");
+
+            // update languageSpecificPrimitives for type hints
+            languageSpecificPrimitives.remove("List");
+            languageSpecificPrimitives.remove("Dict");
+            languageSpecificPrimitives.remove("Set");
+            languageSpecificPrimitives.add("list");
+            languageSpecificPrimitives.add("dict");
+            languageSpecificPrimitives.add("set");
+
+            additionalProperties.put("typingList", "list");
+            additionalProperties.put("typingDict", "dict");
+            additionalProperties.put("typingMap", "map");
+            setSupportPython39(true);
+        } else {
+            additionalProperties.put("typingList", "List");
+            additionalProperties.put("typingDict", "Dict");
+            additionalProperties.put("typingMap", "Map");
```

Comment:
Can we change Map to Dict here?

Reply:
I can do that but can you elaborate on why?

Also when developers see `typingMap` mustache tag in the templates, it would be counter-intuitive to them the tag value is actually "Dict".

If `Map` needs to be mapped to `Dict`, I think that should be done in typeMapping instead

Reply:
There is no Map class in python. Dict is the class that holds maps. Or did you mean ChainMap?

Reply:
Sorry. My mistake. It should be `Mapping` instead. Something like
```
from typing import Mapping, MutableMapping, Sequence, Iterable, List, Set
```

But since we're not using it in our generators/templates, I'll remove it.